drivers/synopsys/usb/dw_usb.c

```diff
@@ -0,0 +1,788 @@
+/*
```

Comment:
Is the datasheet for this device publically avaiable, and does it allow distribution of an open source driver? I have the same IP concerns with this as the designware eMMC driver.